src/Columns/ColumnArray.cpp

```diff
@@ -42,29 +43,34 @@ namespace ErrorCodes
 static constexpr size_t max_array_size_as_field = 1000000;
 
 
-ColumnArray::ColumnArray(MutableColumnPtr && nested_column, MutableColumnPtr && offsets_column)
+ColumnArray::ColumnArray(MutableColumnPtr && nested_column, MutableColumnPtr && offsets_column, bool check_offsets)
```

Comment:
When do we need `check_offsets=false`? This check is not heavy, don't see why we need to skip it.

Reply:
We only need to skip checking offsets in `insertManyFromTuple` and `insertManyFromNullable` because array offsets are appended in advance, it should not be appended inside `insertManyFromTuple` and `insertManyFromNullable`. Setting check_offsets = false makes sure that constructing temporary ColumnArray do not throw exceptions.